fastai/vision/augment.py

```diff
@@ -21,16 +21,31 @@
 class RandTransform(DisplayedTransform):
     "A transform that before_call its state at each `__call__`"
     do,nm,supports,split_idx = True,None,[],0
-    def __init__(self, p=1., nm=None, before_call=None, **kwargs):
+    def __init__(
+        self,
+        p:float=1., # Probability that Transform'd be applied
+        nm:str=None,
+        before_call=None, # Set up function that executed before __call__
+        **kwargs
+    ):
         store_attr('p')
         super().__init__(**kwargs)
         self.before_call = ifnone(before_call,self.before_call)
 
-    def before_call(self, b, split_idx):
+    def before_call(
+        self,
+        b,
+        split_idx:int, # Index of the dataset, if matches with self.idx Transform will be applied
```

Comment:
These `split_idx` are a bit tricky, but this one doesn't change whether the transform is applied
```suggestion
        split_idx:int, # Index of dataset when called
```

Reply:
@warner-benjamin Thanks. I was thinking too much about it and forgot that it has no effect for transform in `before_call`. However, as you said, `split_idx` is quite tricky so I think maybe we need a new design for it. It's hard to guess that changing the split_idx can have an effect on the Transform